iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -163,6 +167,26 @@ public void commitTask(TaskAttemptContext originalContext) throws IOException {
               LOG.info("CommitTask found no serialized table in config for table: {}.", output);
             }
           }, IOException.class);
+
+      // Merge task has merged several files into one. Hence we need to remove the stale files.
+      // At this stage the file is written and task-committed, but the old files are still present.
+      if (CombineHiveInputFormat.class.equals(jobConf.getInputFormat().getClass())) {
+        MapWork mrwork = Utilities.getMapWork(jobConf);
+        if (mrwork != null) {
+          List<Path> mergedPaths = mrwork.getInputPaths();
+          if (CollectionUtils.isNotEmpty(mergedPaths)) {
+            Tasks.foreach(mergedPaths)
+                    .retry(3)
+                    .stopOnFailure()
+                    .throwFailureWhenFinished()
```

Comment:
should we fail if cleanup throws an exception?

Reply:
I think we should not fail. Removed `stopOnFailure` and `throwFailureWhenFinished`, so as to not fail upon failures.